### DISCUSSION TITLE
After building from source: running and configuring milvus (docker)

### DISCUSSION THREAD

**Discussion by benbo** [milvus-io/milvus#23420] [Q&A and General discussion] [answered]
I built milvus from source because I want to use a `DISKANN` index. I managed to build it with `make disk_index=ON` and I started milvus standalone as follows:
```bash
# in the dir where I compiled milvus
cd deployments/docker/standalone
sudo docker compose up -d
```
I was able to create a DISKANN index, add vectors and do a search, everything seems to work well. 

My question now is, how do I configure milvus when I start it via `deployments/docker/standalone`? Does it automatically read the config settings in `configs/milvus.yaml`? The reason I ask is because I want to adapt the DISKANN parameters and change the location of the volume.

**Comment by yhmo** [comment, marked as answer]
The deployments/docker/standalone pull images from docker hub, it doesn't read the local config file `configs/milvus.yaml`. To use docker-compose, you need to build a milvus image and upload to docker hub, then change the image name in the `deployments/docker/standalone/docker-compose.yml`

If you just want to debug milvus, follow the steps:
1. build milvus from source successfully by "make" command under the repo folder
2.  `cd deployments/docker/dev`
3. start the third-parties services(etcd/minio/pulsar) by command: `docker-compose up -d `
4. back to the repo folder by command `cd - `
5. use text tool to modify the `configs/milvus.yaml`, replace all the "/var/lib/milvus" to be "/tmp/milvus"
6. use this script to start a local milvus instance  `./scripts/start_cluster.sh`
7. use this script to stop milvus instance  `./script/stop.sh`

The local milvus instance will read the `configs/milvus.yaml` as configuration, you can change the location of volume in this file before you launching the milvus instance.

**Comment by benbo** [comment]
That makes sense, thank you! So it appears that building a DISKANN index with the image from docker hub works. That was unexpected. 

In order to change the configuration for the image that is pulled from docker hub, I can map the path of a local config file to the corresponding docker container paths under the `volumes:` field (in the `docker-compose.yml` ), correct?

**Reply by yhmo** [reply]
To change the configurations, the online document is: https://milvus.io/docs/configure-docker.md
1. mapping the milvus.yaml path from inside the container to outside, add a line in the volumes field of docker-compose.yml
2. change the items in the outside file milvus.yaml, start the milvus

**Reply by benbo** [reply]
Thank you.  Is there a simple way to check if the changes to the config have been successfully applied (i.e. that linking the outside config was successful)? 

Using the python SDK, `print([index.params for index in collection.indexes])` will unfortunately not list the parameters set in the yml config.

**Reply by yhmo** [reply]
> Thank you. Is there a simple way to check if the changes to the config have been successfully applied (i.e. that linking the outside config was successful)?
> 
> Using the python SDK, `print([index.params for index in collection.indexes])` will unfortunately not list the parameters set in the yml config.

Currently no simple way. You can log into the container or pod to verify the milvus.yml under the /milvus/configs path.

**Reply by benbo** [reply]
Perfect, that helps. Thank you.
